building_sim_plugins/building_ignition_plugins/src/crowd_simulator.cpp

```diff
@@ -76,192 +80,167 @@ void CrowdSimulatorPlugin::PreUpdate(
+          // config internal spawned agent for custom trajectory
+          if (!obj_ptr->is_external)
+          {
+            _config_spawned_agents(obj_ptr, entity, ecm, crowd_sim_interface);
+          }
+          objects_name.erase(name->Data());
+          RCLCPP_INFO(crowd_sim_interface->logger(),
+          "Crowd Simulator found agent: " + name->Data() );
+        }
+        return true;
+      }
+      );
 
-//===================================================================
-bool CrowdSimulatorPlugin::_create_entity(
-  ignition::gazebo::EntityComponentManager& ecm,
-  const std::string& model_name,
-  const crowd_simulator::ModelTypeDatabase::RecordPtr model_type_ptr) const
-{
-  // Use ignition create service to spawn actors
-  // calling ignition gazebo create service, you can use "ign service -l" to check the service available
-  assert(model_type_ptr);
-  std::string service = "/world/" + this->_world_name + "/create";
-  ignition::msgs::EntityFactory request;
-  request.set_sdf_filename(model_type_ptr->file_name);
-  request.set_name(model_name);
-  ignition::math::Pose3d pose(0, 0, 0, 0, 0, 0);
-
-  ignition::msgs::Boolean response;
-  bool result;
-  uint32_t timeout = 5000;
-  bool executed = this->_transport_node_ptr->Request(service, request, timeout,
-      response, result);
-  if (executed)
-  {
-    if (result && response.data())
-    {
-      RCLCPP_INFO(_crowd_sim_interface->logger(),
-        "Requested creation of entity: " + model_name);
-      return true;
-    }
-    else
+    // external agents not found or not loaded yet
+    if (objects_name.size() != 0)
     {
-      RCLCPP_ERROR(_crowd_sim_interface->logger(),
-        "Failed request to create entity.\n" + request.DebugString());
+      _initialized = false;
+      return;
     }
+    _initialized = true;
+    RCLCPP_INFO(
+      crowd_sim_interface->logger(),
+      "Ignition Models are all loaded! Start simulating...");
   }
-  else
-  {
-    RCLCPP_ERROR(
-      _crowd_sim_interface->logger(),
-      "Request to create entity from service " + service + "timer out ...\n" +
-      request.DebugString());
-  }
-  return false;
 }
 
 //==================================================
 void CrowdSimulatorPlugin::_config_spawned_agents(
   const crowd_simulator::CrowdSimInterface::ObjectPtr obj_ptr,
   const ignition::gazebo::Entity& entity,
-  ignition::gazebo::EntityComponentManager& ecm) const
+  ignition::gazebo::EntityComponentManager& ecm,
+  std::shared_ptr<crowd_simulator::CrowdSimInterface> crowd_sim_interface) const
 {
-  assert(obj_ptr);
-  auto agent_ptr = obj_ptr->agent_ptr;
-  auto model_type = _crowd_sim_interface->_model_type_db_ptr->get(
+  bool disabled = crowd_sim_interface->_menge_disabled;
+  assert(disabled || obj_ptr);
+  auto agent_ptr = (disabled) ? nullptr : obj_ptr->agent_ptr;
+  auto model_type =
+    (disabled) ? nullptr : crowd_sim_interface->_model_type_db_ptr->get(
```

Comment:
I can change the check to be on the pointers (they are only non-null when menge is loaded). Removing ternary operator would add few more lines instead of one-liner.